shotover-proxy/tests/kafka_int_tests/mod.rs

```diff
@@ -247,21 +247,55 @@ async fn single_sasl_scram_plaintext_source_tls_sink(#[case] driver: KafkaDriver
 #[case::java(KafkaDriver::Java)]
 #[tokio::test(flavor = "multi_thread")] // multi_thread is needed since java driver will block when consuming, causing shotover logs to not appear
 async fn cluster_1_rack_single_shotover(#[case] driver: KafkaDriver) {
-    let _docker_compose =
+    let docker_compose =
         docker_compose("tests/test-configs/kafka/cluster-1-rack/docker-compose.yaml");
-    let shotover = shotover_process("tests/test-configs/kafka/cluster-1-rack/topology-single.yaml")
-        .start()
-        .await;
 
-    let connection_builder = KafkaConnectionBuilder::new(driver, "127.0.0.1:9192");
-    test_cases::cluster_test_suite(&connection_builder).await;
+    {
+        let shotover =
+            shotover_process("tests/test-configs/kafka/cluster-1-rack/topology-single.yaml")
+                .start()
+                .await;
 
-    tokio::time::timeout(
-        Duration::from_secs(10),
-        shotover.shutdown_and_then_consume_events(&[]),
-    )
-    .await
-    .expect("Shotover did not shutdown within 10s");
+        let connection_builder = KafkaConnectionBuilder::new(driver, "127.0.0.1:9192");
+        test_cases::cluster_test_suite(&connection_builder).await;
+        tokio::time::timeout(
+            Duration::from_secs(10),
+            shotover.shutdown_and_then_consume_events(&[]),
+        )
+        .await
+        .expect("Shotover did not shutdown within 10s");
+    }
+
+    {
+        let shotover =
+            shotover_process("tests/test-configs/kafka/cluster-1-rack/topology-single.yaml")
+                .start()
+                .await;
+        let connection_builder = KafkaConnectionBuilder::new(driver, "127.0.0.1:9192");
+
+        test_cases::produce_consume_partitions1_kafka_node_goes_down(
+            driver,
+            &docker_compose,
+            &connection_builder,
+            "kafka_node_goes_down_test",
+        )
+        .await;
+
+        // Shotover can reasonably hit many kinds of errors due to a kafka node down so ignore all of them.
```

Comment:
Is it possible to specify the expected error types here so that we can still catch errors that shouldn't happen during a kafka node down?

Reply:
There are a lot of errors that can occur during a kafka node being down so I dont think its really feasible at the moment.
The test logic itself still guarantees that all the kafka functionality is upheld, so I'm not too worried.

I think there would be value in extending our [test infrastructure](https://github.com/shotover/tokio-bin-process) to support matching errors on a regex, but that is out of scope of this work.